src/github.com/getlantern/flashlight/client/balancer.go

```diff
@@ -14,9 +15,18 @@ func (client *Client) initBalancer(cfg *ClientConfig) *balancer.Balancer {
 	dialers := make([]*balancer.Dialer, 0, len(cfg.FrontedServers)+len(cfg.ChainedServers))
 
 	log.Debugf("Adding %d domain fronted servers", len(cfg.FrontedServers))
+	var highestQOSFrontedDialer fronted.Dialer
+	highestQOS := -1
 	for _, s := range cfg.FrontedServers {
-		dialer := s.dialer(cfg.MasqueradeSets)
+		fd, dialer := s.dialer(cfg.MasqueradeSets)
 		dialers = append(dialers, dialer)
+		if dialer.QOS > highestQOS {
+			highestQOSFrontedDialer = fd
+		}
+	}
+
+	if highestQOSFrontedDialer != nil {
+		go determinePublicIp(highestQOSFrontedDialer)
 	}
```

Comment:
Yeah, let's cross that bridge when we come to it.  It's still not 100% clear whether we'll even have the same program for give and get mode.

@xiam This reminds me, we need more than just the public IP.  We need the geolocation information (lat/lon) to place the user's dot on the map.  That information is available in the body of the response.  You can use [this](https://github.com/getlantern/flashlight-build/blob/valencia/src/github.com/getlantern/flashlight/statserver/peer.go#L40) data structure to capture the JSON from the body.  You'll probably want to create a new geo package inside flashlight or something like that and then share it between client and statreporter.